### PR TITLE
telegram: truncate message to 4096 length to avoid destination getting stuck

### PR DESCRIPTION
**Pull request by furiel**
Telegram returns with 400 http response with `MESSAGE_TOO_LONG`, if the telegram message is longer than 4096 utf8 characters. I could not find this value in the official documentation, but unofficially I could find it here and there. I also tested, and 4096 long message passes, but 4097 fails. The request itself can be longer (we send other fields). Only the message field is restricted to 4096.

This patchset adds to telegram destination an additional parameter: `truncate` which tells the truncate length, which will be applied to the message. Initial value is 4096.

I wish there were `maximum` template function, that I could apply, if users provide larger length than 4096. (There is a `max` template function, but that works on message contexts instead of raw numbers. I will create a good first issue feature request for it).

Since 3.25, syslog-ng disconnects for 400 by default. It means the problematic message is put back to the queue, then retried after time-reopen, infinitely blocking telegram notifications.

Prior to 3.26, the 400 error code was mapped to drop, instead of disconnect, so in those versions only the message is lost, but the connection did not get stuck. I was thinking if we may change the default action of 400 of telegram destination, but the other 400-related events seems fine for disconnect. https://core.telegram.org/method/messages.sendMessage

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by alltilla** [comment]
The name `truncate()` seems like a `yesno` kind of option for me.

I believe `truncate-at()` or `max-payload-length()`/`max-message-length()` might fit better, just a suggestion, though.

**Comment by furiel** [comment]
I changed `truncate` to `max-size`.

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: news/feature-3206.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+`telegram`: new `truncate` option
```
Comment (alltilla) [inline news/feature-3206.md]: Please update the option name here, too.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**